### PR TITLE
Added lesson 08 solutions

### PR DESCRIPTION
**Pull request by swietmon**
@kamwar please check

### CODE REVIEW

File: students/swietczak_monika/lesson_08_regular_expressions/email_validator.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+def email_validator(some_email):
+    if regex.match(some_email) is None:
+        print("email is not valid")
+    else:
+        print("email is valid")
+
+
+# email_validator("mail123@test.com")
+# email_validator("mail.mail")
+# email_validator("mail@test")
```
Comment (kamwar) [inline students/swietczak_monika/lesson_08_regular_expressions/email_validator.py]: Please consider to add py.test for verification.

There is available perfect feature (parametrize) for calling the same method many times.

```
class TestEmailAddressValidator:
    @pytest.mark.parametrize("correct_address", [
        'aaa@aaa.pl',
        'a-a-a@aaa.pl',
        'a12a@12a.com',
    ])
    def test_valid_email_address(self, correct_address):
        assert email_validator(correct_address)

    @pytest.mark.parametrize("incorrect_address", [
        'aaa@a@aa.pl',
        'aaa@@aa.pl',
        'a12a.com',
    ])
    @pytest.mark.xfail
    def test_incorrect_email_address(self, incorrect_address):
        assert email_validator(incorrect_address)
```

File: students/swietczak_monika/lesson_08_regular_expressions/phone_number_validator.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import re
+
+regex = re.compile(r"^\d\d\d[- ]?\d\d\d[- ]?\d\d\d$")
```
Comment (kamwar) [inline students/swietczak_monika/lesson_08_regular_expressions/phone_number_validator.py]: Consider to use repeated qualifier

```
\d\d\d -> \d{3}
```

File: students/swietczak_monika/lesson_08_regular_expressions/regex_version_of_strip.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import re
+
+
+def regex_strip(some_string, chars_to_strip):
+    if re.search(r"[\.\*\\]", chars_to_strip) is not None:
+        result = re.sub("^" + re.escape(chars_to_strip) + "*", '', some_string)
+        result = re.sub(re.escape(chars_to_strip) + "*$", '', result)
+    else:
+        result = re.sub(r'^' + chars_to_strip + '*', '', some_string)
+        result = re.sub(chars_to_strip + '*$', '', result)
```
Comment (kamwar) [inline students/swietczak_monika/lesson_08_regular_expressions/regex_version_of_strip.py]: You might use OR statement for above two cases:

regex = re.compile(r'^[' + chars_to_strip + ']*|[' + chars_to_strip + ']*$')
re.sub(regex, '', string)

**Comment by swietmon** [comment]
Modifications done @kamwar

File: students/swietczak_monika/lesson_08_regular_expressions/email_validator.py [hunk 4]
```diff
@@ -1,19 +1,35 @@
...
 
+# user_email = input("Please enter the email address: ")
+# email_validator(user_email)
 
-# email_validator("mail123@test.com")
-# email_validator("mail.mail")
-# email_validator("mail@test")
 
+class TestEmailAddressValidator:
+    @pytest.mark.parametrize("correct_address", [
```
Comment (kamwar) [inline students/swietczak_monika/lesson_08_regular_expressions/email_validator.py]: I know it wasn't introduced in the course but it looks more robust now :)

**Pull request merged**